bundles/org.openhab.binding.viessmann/src/main/java/org/openhab/binding/viessmann/internal/handler/DeviceHandler.java

```diff
@@ -0,0 +1,1165 @@
+                            if (feature.contains("oneTimeCharge")) {
+                                initStateDelay = 2;
+                                initState = true;
+                            }
+                        }
+                    } else if (OnOffType.OFF.equals(command)) {
+                        uri = prop.get("deactivateUri");
+                        param = "{}";
+                        String feature = prop.get("feature");
+                        if (feature != null) {
+                            if (feature.contains("oneTimeCharge")) {
+                                initStateDelay = 2;
+                                initState = true;
+                            }
+                        }
+                    } else if (command instanceof DecimalType) {
+                        logger.trace("Received DecimalType Command for Channel {}",
+                                thing.getChannel(channelUID.getId()));
+                        for (String str : com) {
+                            if (str.contains("setCurve")) {
+                                uri = prop.get(str + "Uri");
+                                String circuitId = prop.get("circuitId");
+                                HeatingCircuit heatingCircuit = heatingCircuits.get(circuitId);
+                                if (heatingCircuit != null) {
+                                    String slope = heatingCircuit.getSlope();
+                                    String shift = heatingCircuit.getShift();
+                                    String value = command.toString();
+                                    if (ch.getUID().toString().contains("shift")) {
+                                        param = "{\"slope\":" + slope + ", \"shift\":" + value + "}";
+                                        heatingCircuit.setShift(value);
+                                    } else if (ch.getUID().toString().contains("slope")) {
+                                        param = "{\"slope\":" + value + ", \"shift\":" + shift + "}";
+                                        heatingCircuit.setSlope(value);
+                                    }
+                                    if (circuitId != null) {
+                                        heatingCircuits.put(circuitId, heatingCircuit);
+                                    }
+                                }
+                                break;
+                            }
+                            if (str.contains("setHysteresis")) {
+                                String f = command.toString();
+                                uri = prop.get(str + "Uri");
+                                param = "{\"" + prop.get(str + "Params") + "\":" + f + "}";
+                                break;
+                            }
+
+                        }
+                    } else if (command instanceof QuantityType<?>) {
+                        QuantityType<?> value = (QuantityType<?>) command;
```

Comment:
Java 17 instanceof pattern matching can be used on more places
```suggestion
                    } else if (command instanceof QuantityType<?> value) {
```